flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-oceanbase/src/main/java/org/apache/flink/cdc/connectors/oceanbase/sink/OceanBaseUtils.java

```diff
@@ -0,0 +1,244 @@
+ */
+
+package org.apache.flink.cdc.connectors.oceanbase.sink;
+
+import org.apache.flink.cdc.common.event.TableId;
+import org.apache.flink.cdc.common.schema.Column;
+import org.apache.flink.cdc.common.schema.Schema;
+import org.apache.flink.cdc.common.types.BigIntType;
+import org.apache.flink.cdc.common.types.BooleanType;
+import org.apache.flink.cdc.common.types.CharType;
+import org.apache.flink.cdc.common.types.DataType;
+import org.apache.flink.cdc.common.types.DataTypeDefaultVisitor;
+import org.apache.flink.cdc.common.types.DateType;
+import org.apache.flink.cdc.common.types.DecimalType;
+import org.apache.flink.cdc.common.types.DoubleType;
+import org.apache.flink.cdc.common.types.FloatType;
+import org.apache.flink.cdc.common.types.IntType;
+import org.apache.flink.cdc.common.types.LocalZonedTimestampType;
+import org.apache.flink.cdc.common.types.SmallIntType;
+import org.apache.flink.cdc.common.types.TimestampType;
+import org.apache.flink.cdc.common.types.TinyIntType;
+import org.apache.flink.cdc.common.types.VarCharType;
+import org.apache.flink.cdc.connectors.oceanbase.catalog.OceanBaseColumn;
+import org.apache.flink.cdc.connectors.oceanbase.catalog.OceanBaseTable;
+
+import org.apache.commons.collections.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** Utilities for conversion from source table to OceanBase table. */
+public class OceanBaseUtils {
+
+    /** Convert a source table to {@link OceanBaseTable}. */
+    public static OceanBaseTable toOceanBaseTable(TableId tableId, Schema schema) {
+
+        List<Column> columns = schema.getColumns();
+        List<String> primaryKeys = schema.primaryKeys();
+        List<OceanBaseColumn> oceanBaseColumns = new ArrayList<>();
+        for (int i = 0; i < columns.size(); i++) {
+            Column column = columns.get(i);
+            OceanBaseColumn.Builder builder =
+                    new OceanBaseColumn.Builder()
+                            .setColumnName(column.getName())
+                            .setOrdinalPosition(i)
+                            .setColumnComment(column.getComment());
+            toOceanBaseDataType(column, primaryKeys.contains(column.getName()), builder);
+            oceanBaseColumns.add(builder.build());
+        }
+
+        OceanBaseTable.Builder tableBuilder =
+                new OceanBaseTable.Builder()
+                        .setDatabaseName(tableId.getSchemaName())
+                        .setTableName(tableId.getTableName())
+                        .setTableType(
+                                CollectionUtils.isNotEmpty(schema.primaryKeys())
+                                        ? OceanBaseTable.TableType.PRIMARY_KEY
+                                        : OceanBaseTable.TableType.DUPLICATE_KEY)
+                        .setColumns(oceanBaseColumns)
+                        .setTableKeys(schema.primaryKeys())
+                        .setPartitionKeys(schema.partitionKeys())
```

Comment:
- Since the connector does not yet support OceanBase partitioned tables, first make partitionKeys part of primaryKeys.
- I added a check that the primary key is not allowed to be null in oceanbase.